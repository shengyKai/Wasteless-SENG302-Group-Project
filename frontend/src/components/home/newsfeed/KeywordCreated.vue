<template v-if="showEvent">
  <Event :event="event" :title="title">
    <v-card-text>
      {{ creator.firstName }} {{ creator.lastName }} has added "{{ keyword.name }}"
    </v-card-text>
    <v-card-actions class="justify-center">
      <v-btn color="error" @click="showDeleteDialog=true">Remove</v-btn>
    </v-card-actions>
    <template v-if="showDeleteDialog">
      <DeleteKeyword :keyword="keyword" @closeDialog="showDeleteDialog=false" @keywordDeleted="showEvent=false"/>
    </template>
  </Event>
</template>

<script>
import Event from './Event';
import DeleteKeyword from "../../admin/DeleteKeyword.vue";

export default {
  name: 'KeywordCreated',
  data() {
    return {
      showDeleteDialog: false,
      showEvent: true,
    };
  },
  props: {
    event: {
      type: Object,
    },
  },
  components: {
    Event,
    DeleteKeyword,
  },
  computed: {
    keyword() {
      return this.event.keyword;
    },
    creator() {
      return this.event.creator;
    },
    title() {
      return `Keyword "${this.keyword.name}" has been created`;
    },
  },
};
</script>