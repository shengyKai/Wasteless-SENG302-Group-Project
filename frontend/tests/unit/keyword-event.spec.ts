
import Vue from 'vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';
import KeywordCreated from '@/components/home/newsfeed/KeywordCreated.vue';

Vue.use(Vuetify);

describe('KeywordCreated.vue', () => {
  let wrapper: Wrapper<any>;
  let vuetify: Vuetify;
  beforeEach(async () => {
    const localVue = createLocalVue();
    vuetify = new Vuetify();

    const app = document.createElement ("div");
    app.setAttribute ("data-app", "true");
    document.body.append (app);

    wrapper = mount(KeywordCreated, {
      localVue,
      vuetify,
      propsData: {
        event: {
          id: 101,
          created: "2021-01-01T12:00:00Z",
          type: "KeywordCreatedEvent",
          keyword: {
            name: 'TestKeywordName',
          },
          creator: {firstName: 'John'  , lastName: 'Smith'},
        },
      }
    });
  });

  it("Title is correct", () => {
    let eventWrapper = wrapper.findComponent({ name: 'Event'});
    expect(eventWrapper.props().title).toBe('Keyword "TestKeywordName" has been created');
  });

  it('Contains creator\'s name', () => {
    expect(wrapper.text()).toContain('John Smith');
  });

  it('Must match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});