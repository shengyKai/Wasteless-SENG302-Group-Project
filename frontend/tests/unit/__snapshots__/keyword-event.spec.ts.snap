// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeywordCreated.vue Must match snapshot 1`] = `
<div>
  <div class="delete">
    <div class="float=right ma-1"><button type="button" class="v-icon notranslate deleteButton v-icon--link mdi mdi-trash-can theme--light red--text"></button></div>
    <div class="v-card__title">
      Keyword "TestKeywordName" has been created
    </div>
    <div class="v-card__subtitle">
      02 Jan 2021, 01:00
    </div>
    <div class="v-card__text">
      John Smith has added "TestKeywordName"
    </div>
    <div class="v-card__actions justify-center"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default error"><span class="v-btn__content">Remove</span></button></div>
    <!---->
    <!---->
    <div class="row">
      <div class="col"><span class="ma-2 ml-4 mb-3 v-chip v-chip--clickable v-chip--label v-chip--no-color theme--light v-size--default white--text" aria-haspopup="true" aria-expanded="false"><span class="v-chip__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-label theme--light"></i>
              Tag
            </span></span><span class="v-tooltip v-tooltip--bottom"><!----></span></div>
    </div>
    <div class="row">
      <div class="col">
        <transition-stub name="expand-transition">
          <div class="mx-auto v-card v-sheet theme--light" style="max-width: 1200px; display: none;">
            <div class="font-weight-medium"><span class="ml-4">
                Change your Tag:
              </span></div>
            <div class="ml-3"><span class="ma-1 v-chip v-chip--clickable v-chip--label theme--light v-size--default none white--text"><span class="v-chip__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-label theme--light"></i></span></span><span class="ma-1 v-chip v-chip--clickable v-chip--label theme--light v-size--default red white--text"><span class="v-chip__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-label theme--light"></i></span></span><span class="ma-1 v-chip v-chip--clickable v-chip--label theme--light v-size--default orange white--text"><span class="v-chip__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-label theme--light"></i></span></span><span class="ma-1 v-chip v-chip--clickable v-chip--label theme--light v-size--default yellow white--text"><span class="v-chip__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-label theme--light"></i></span></span><span class="ma-1 v-chip v-chip--clickable v-chip--label theme--light v-size--default green white--text"><span class="v-chip__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-label theme--light"></i></span></span><span class="ma-1 v-chip v-chip--clickable v-chip--label theme--light v-size--default blue white--text"><span class="v-chip__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-label theme--light"></i></span></span><span class="ma-1 v-chip v-chip--clickable v-chip--label theme--light v-size--default purple white--text"><span class="v-chip__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-label theme--light"></i></span></span></div>
          </div>
        </transition-stub>
      </div>
    </div>
  </div>
</div>
`;
